### PR TITLE
Add CI setup for django test suite

### PR DESCRIPTION
**Pull request by Sfurti-yb**

### CODE REVIEW

File: .github/workflows/django_test.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
...tests logging_tests m2o_recursive mail max_lengths messages_tests middleware middleware_exceptions migrate_signals migrations.test_autodetector migrations.test_commands.AppLabelErrorTests migrations.test_loader migrations.test_commands.MakeMigrationsTests migrations.test_commands.OptimizeMigrationTests migrations.test_commands.SquashMigrationsTests migrations.test_exceptions migrations.test_executor migrations.test_graph migrations.test_multidb migrations.test_operations.FieldOperationTests migrations.test_operations.SwappableOperationTests migrations.test_optimizer migrations.test_questioner migrations.test_state migrations.test_writer migration_test_data_persistence migrations2 model_enums model_indexes model_meta model_utils no_models or_lookups pagination project_template properties requests reserved_names resolve_url responses save_delete_hooks servers sessions_tests settings_tests shell shortcuts signed_cookies_tests signing sitemaps_tests sites_tests staticfiles_tests str template_backends template_loader test_client test_client_regress test_exceptions test_runner transaction_hooks transactions urlpatterns urlpatterns_reverse user_commands utils_tests validators version wsgi
```
Comment (felixxm) [inline .github/workflows/django_test.yml:26]: I'd recommend to run all `migrations` and `schema` tests and list tests that don't work in `django_test_expected_failures`/`django_test_skips` with appropriate comments. See how it'd handled in [django-cockroachdb](https://github.com/cockroachdb/django-cockroachdb/blob/46df2730abe8fcdc1e4b0370023eb174b058331f/django_cockroachdb/features.py#L89-L210).

Comment (Sfurti-yb) [reply .github/workflows/django_test.yml]: I have updated the file now to run both `migrations` and `schema` tests.

**Comment by felixxm** [comment]
> @felixxm Gentle reminder to take a look at this.

If your backends supports Django 3.1+ you should test all versions 3.1, 3.2, and 4.0. Also, it's probably not worth testing/supporting Django 3.1 which is EOL.

**Comment by Sfurti-yb** [comment]
Right now, the CI setup pulls the latest Django version (4.0) and checks against it. Should we add checks for versions 3.2 then as well?

**Comment by felixxm** [comment]
> Right now, the CI setup pulls the latest Django version (4.0) and checks against it. Should we add checks for versions 3.2 then as well?

If you want to support it, then yes.

**Comment by Sfurti-yb** [comment]
Okay, we will add checks for 3.2 as well. Any other changes you suggest before approving the PR?

**Comment by felixxm** [comment]
Currently you're testing with Django 3.2 and the current `main` branch, so Django 4.0 and 4.1 are still untested. Can you use a single generic script and pass a Django version to it? or move bash scripts' logic to the GA configuration?

**Comment by Sfurti-yb** [comment]
@felixxm I have added support for 4.0 and 4.1. Changed multiple bash scripts to single. Please take a look.

**Review by felixxm — APPROVED** [review]

**Pull request merged**